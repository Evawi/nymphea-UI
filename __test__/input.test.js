import React from 'react';
import INPUT from '../src/component/input/input'
//import Nya from '../public/nymphea_UI.v_0.0.0'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'


//чтоб обновить снапшоты юзай npm test -- -u  если запустишь просто npm test тест ругнется на новый элемент


describe('>>>INPUT simple',()=>{

    it('+++Snapshot default ', () => {
        const renderedValue =  renderer.create(<INPUT />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
    it('+++Snapshot PROPS popupLabel ', () => {
        const renderedValue =  renderer.create(
            <INPUT
                 popupLabel="label value"
            />
        ).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
    it('+++Snapshot STATE error', () => {
        const input = shallow(
            <INPUT type="number" />
        );
        input.find('input').simulate('change', {target: {value: 'abc'} });
        const renderedValue =  renderer.create(<INPUT />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
    it('---should handle the onChange event', () => {
        const mockFn = jest.fn();
        const input = shallow(
            <INPUT  />
        );
        input.find('input').simulate('change', {target: {value: 'matched'} });
        expect(input.state().value).toEqual('matched');
        expect(input.state().error).toEqual(false);
    });

    it('---should handle the onChange event PROPS checkEmpty', () => {
        const mockFn = jest.fn();
        const input = shallow(
            <INPUT
             checkEmpty={true}
                />
        );
        input.find('input').simulate('change', {target: {value: ''} });
        expect(input.state().value).toEqual('');
        expect(input.state().error).toEqual(true);
    });
    it('---should handle the onChange event PROPS type NUMBER', () => {
        const mockFn = jest.fn();
        const input = shallow(
            <INPUT type="number" />
        );
        input.find('input').simulate('change', {target: {value: '123'} });
        expect(input.state().value).toEqual('123');
        expect(input.state().error).toEqual(false);
    });
    it('---should handle the onChange event PROPS type NUMBER error', () => {
        const mockFn = jest.fn();
        const input = shallow(
            <INPUT type="number" />
        );
        input.find('input').simulate('change', {target: {value: 'abc'} });
        expect(input.state().value).toEqual('');
        expect(input.state().error).toEqual(true);
    });
});