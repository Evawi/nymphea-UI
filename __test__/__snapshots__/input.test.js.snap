// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>INPUT simple +++Snapshot PROPS popupLabel  1`] = `
<div
  className=" nymphea_input "
>
  <div
    className=" nymphea_input__field "
  >
    <div
      className=" nymphea_input__popup_label  bottom center "
      data-position="bottom right"
      data-tooltip="label value"
    >
      <input
        className="nymphea_input__element"
        max=""
        maxLength=""
        min=""
        onChange={[Function]}
        pattern=""
        placeholder=""
        title=""
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`>>>INPUT simple +++Snapshot STATE error 1`] = `
<div
  className=" nymphea_input "
>
  <div
    className=" nymphea_input__field "
  >
    <div
      className=" nymphea_input__popup_label  bottom center "
      data-position="bottom right"
    >
      <input
        className="nymphea_input__element"
        max=""
        maxLength=""
        min=""
        onChange={[Function]}
        pattern=""
        placeholder=""
        title=""
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`>>>INPUT simple +++Snapshot default  1`] = `
<div
  className=" nymphea_input "
>
  <div
    className=" nymphea_input__field "
  >
    <div
      className=" nymphea_input__popup_label  bottom center "
      data-position="bottom right"
    >
      <input
        className="nymphea_input__element"
        max=""
        maxLength=""
        min=""
        onChange={[Function]}
        pattern=""
        placeholder=""
        title=""
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
